### PR TITLE
feat: :sparkles: add request forget password and integration

### PR DESCRIPTION
**Pull request by oliver-mf**
### Background: 
- add forget password request page and its integration
- add customizable paddings to form component

### Additional Packages/Dependencies:
- none

### Changes:
- auth/forget/bloc/forget_cubit.dart
- auth/forget/bloc/forget_state.dart
- auth/forget/data/forget_data_provider.dart
- auth/forget/data/forget_model.dart
- auth/forget/data/forget_model.g.dart
- auth/forget/data/forget_repository.dart
- common/components/forms/form_component.dart
- common/config/route_generator.dart
- page/forget.dart

### CODE REVIEW

File: lib/page/forget.dart [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+      children: [
+        Padding(
+          padding: const EdgeInsets.only(top: 29.0, bottom: 45.0),
+          child: Center(
+              child:
+                  Image.asset('lib/common/assets/images/GlobeIconSmall.png')),
+        ),
+        BlocBuilder<ForgetCubit, ForgetState>(builder: (context, state) {
+          if (state is ForgetSuccessState) {
+            return const Text("success");
```
Comment (YafiAhsanH) [inline lib/page/forget.dart]: I think you need to change the success handler, as of now it will just show success text, might want to redirect the page.

Comment (oliver-mf) [reply lib/page/forget.dart]: alright ive changed it to redirect to another page (though its still a dummy because there's no login page yet)

Comment (YafiAhsanH) [reply lib/page/forget.dart]: nice

**Review by YafiAhsanH — APPROVED** [review]
Looks good!

**Review by RafadanaM — APPROVED** [review]

**Pull request merged**